### PR TITLE
Reorganize signer dependencies

### LINKED ISSUE #1906: Reorganize signer dependencies
## Why
The signer dependencies are not well organized and are not aligned with the aggregator. We need to reorganize them as a preliminary step to the work on the decentralization of the signature orchestration.

## What 
Reorganize signer dependencies by reproducing the structure in place in the aggregator

## How

- [ ] Reorganize signer dependencies

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR reorganize `mithril-signer` source directory based on `mithril-aggregator` current structure:

- `runtime/signer_services`: 
  - Moved to a `dependency_injection` module
  - Renamed types: `ServiceBuilder` -> `DependenciesBuilder`, `SignerService` -> `SignerDependencyContainer`)
  - Types are split into a `builder` and `container` submodule
- Introduce a `services` module:
  - Contains aggregator client, transactions importer, single signer and upkeep service
  - Since transactions importer are split in several files (because of the three decorators) it's itself a submodule directory in the services module.
- Introduce a `store` module that contains the `MKTreeStoreSqlite` and the `ProtocolInitializerStore`

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [ ] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Issue(s)

Closes #1906

### CODE REVIEW

**Review by jpraynaud — APPROVED** [review]
LGTM

I just left few suggestions

File: mithril-signer/src/services/cardano_transactions_importer/mod.rs
[the remark this answers is not in the document]
Comment (Alenar) [reply mithril-signer/src/services/cardano_transactions_importer/mod.rs:1]: I would rather keep a segmentation by tasks for the services. As an alternative I could do the following:
```
services/
|_ cardano_transactions/
    |- mod.rs
    |- preloader.rs
    |_ importer/
        |- mod.rs
        |- service.rs
        |- (...)
```

File: mithril-signer/src/dependency_injection/builder.rs [hunk 1]
```diff
@@ -1,78 +1,79 @@
...
+pub trait DependenciesBuilder {
     /// Create a SignerService instance.
-    async fn build(&self) -> StdResult<SignerServices>;
+    async fn build(&self) -> StdResult<SignerDependencyContainer>;
 }
 
-/// Create a SignerService instance for Production environment.
-pub struct ProductionServiceBuilder<'a> {
+/// A `DependenciesBuilder` for Production environment.
+pub struct ProductionDependenciesBuilder<'a> {
```
[the remark this answers is not in the document]
Comment (Alenar) [reply mithril-signer/src/dependency_injection/builder.rs:60]: This conflict with the trait `DependenciesBuilder` :sweat_smile:  (that this struct implement).

[the remark this answers is not in the document]
Comment (jpraynaud) [reply mithril-signer/src/dependency_injection/builder.rs:60]: Indeed, maybe this trait is not necessary after all?

**Review by sfauvel — APPROVED** [review]
LGTM

**Pull request merged**